department-of-veterans-affairs/caseflow
# Jonathan/caseflow 3972 refactor of pre docket task

Resolves [CASEFLOW-3972](https://vajira.max.gov/browse/CASEFLOW-3972)

### Description
Refactored PreDocketTask to allow the addition of the Education business line

### Acceptance Criteria
- [ ] The PreDocketTask is refactored so that it may be utilized for the Education business line
- [ ] Preexisting functionality for the VHA business line is not negatively impacted in any way

### Testing Plan
Run rspec tests:
![image](https://user-images.githubusercontent.com/98970951/161813248-e86ca996-116f-4426-9ee0-442450921b2e.png)

![image](https://user-images.githubusercontent.com/98970951/161813530-c7d8bad5-abf8-45f0-8685-762f8c833c24.png)

1. In appeal.rb, uncomment the elsif block in create_tasks_on_intake_success! and temporarily remove the "ri.is_predocket_needed " check from edu_predocket_needed? if those changes have not been merged yet.
2. In Caseflow, switch to a user with Intake permissions.
3. Navigate to /intake and begin the 10182 process.
4. When you get to the Add/Remove Issues page, add an issue and select "education" as the benefit type.
5. Click "Establish Appeal".
6. Verify that the appeal went into pre-docket queue.
7. Search for the appeal in Caseflow Queue, verify that it is assigned to "Education" and the status is "Pre Docketed", then navigate to the case details page.
8. On the Case Details page, click on the Actions dropdown menu and confirm that "Return Appeal to Education Service" appears.
9. Grab the uuid from the url.
10. Open the rails console and get the appeal's data using the uuid. Verify that the task tree contains the EducationDocumentSearchTask.

![image](https://user-images.githubusercontent.com/98970951/161817688-3cbaf75e-f1ed-4924-8a5b-b4d3f655e062.png)

### User Facing Changes

 BEFORE|AFTER
 ---|---
![image](https://user-images.githubusercontent.com/98970951/162041962-dae0a882-e9f8-4709-b554-62d7ac3f753e.png) | ![image](https://user-images.githubusercontent.com/98970951/162050235-279886d0-4ac7-4392-8a91-cfe0602859ba.png) |
![image](https://user-images.githubusercontent.com/98970951/162050351-b2c2707d-9e13-4500-afe5-eb0188e09c2e.png) | ![image](https://user-images.githubusercontent.com/98970951/162050476-70676fdb-7329-4d0a-8a01-2b4e3cd88d31.png)
